### PR TITLE
[opentitanlib] Fix of rust code formatting

### PR DESCRIPTION
**Pull request by semihalf-mazurek-michal**
Fix code formatting by applying cargo fmt over opentitantools.
Add cargo fmt invocation to meson build script.

Signed-off-by: Michał Mazurek <maz@semihalf.com>

### CODE REVIEW

**Review by mcy — COMMENTED** [review]
I don't see any changes to CI. How do we plan to prevent backsliding? @milesdai @drewmacrae

**Comment by timothytrippel** [comment]
Thanks for this @semihalf-mazurek-michal ! As @mcy mentioned, I think we are going to want to add CI script like [this](https://github.com/lowRISC/opentitan/blob/master/ci/scripts/clang-format.sh) to this PR, but for rust formatting.  Then you will have to format all rust code in the repo in this PR as well, for CI to pass, but that should just be everything in `sw/host`. Could you file an issue on this? and add the label "Hotlist:Software"? Then we can bring this up in the SW Working group next week and get this resolved.

**Comment by semihalf-mazurek-michal** [comment]
@timothytrippel  I added following issue: #11108

**Comment by semihalf-mazurek-michal** [comment]
@timothytrippel @mcy I update this PR with solution to prevent backsliding.
This solution adds cargo fmt to current build scripts.
It's not a perfect solution, but it should be enough for now.

File: sw/host/opentitantool/meson.build [hunk 1]
```diff
@@ -22,6 +22,8 @@ foreach flag : cargo_flags_array
   cargo_flags += '--' + flag + ' '
 endforeach
 
+cargo_fmt_flag = '--manifest-path=' + manifest_path
```
Comment (semihalf-mazurek-michal) [inline sw/host/opentitantool/meson.build:25]: Flag --check could be added to force an error when compiling unformatted code.

File: util/invoke_cargo.sh [hunk 2]
```diff
@@ -11,20 +11,23 @@ set -e
...
 fi
 
-export MESON_SOURCE_ROOT="${5}"
-export MESON_BUILD_ROOT="${6}"
+export MESON_SOURCE_ROOT="${6}"
+export MESON_BUILD_ROOT="${7}"
 
 if [ "${CARGO_TEST}" == 1 ]; then
     echo "CARGO TEST BUILD!"
+    "${CARGO}" fmt ${CARGO_FMT_FLAG}
```
Comment (timothytrippel) [inline util/invoke_cargo.sh:27]: @mcy Thoughts on calling cargo format in this script that is invoked by meson?

**Review by timothytrippel — COMMENTED** [review]
@semihalf-mazurek-michal Thanks for submitting this, the SW team agrees we should check rust formatting. While adding the invocation of `cargo fmt` to the script that meson invokes when building rust crates is a way to enable developers to automatically format rust code when they build it locally, it does not enforce a CI check, i.e., someone could submit a PR without attempting to build the rust code locally, and it would not be formatted.

In addition to these changes, you will need to add a script similar to the [`clang-format.sh`](https://github.com/lowRISC/opentitan/blob/master/ci/scripts/clang-format.sh) script in CI and make sure it is invoked from the CI [`quick-lint.sh`](https://github.com/lowRISC/opentitan/blob/master/ci/jobs/quick-lint.sh) script (again, [similar to how `clang-format.sh` is invoked in this script](https://github.com/lowRISC/opentitan/blob/b4e64196eaf1d3844b27e5ac7e5ddeebccb50236/ci/jobs/quick-lint.sh#L54)). Let me know if you have any questions!

File: util/invoke_cargo.sh (hunk 2 above)
Comment (mcy) [reply util/invoke_cargo.sh:27]: Sure, not wrong. Can this be split off into its own commit, sequenced after applying the formatting cleanup?

**Review by mcy — APPROVED** [review]
Approved modulo breaking this into two commits.

File: util/invoke_cargo.sh (hunk 2 above)
Comment (semihalf-mazurek-michal) [reply util/invoke_cargo.sh:27]: I will split it for 2 commits:
1) formatting cleanup
2) scripts with cargo fmt

**Comment by timothytrippel** [comment]
There were some CI issues earlier today, that were recently fixed. just rebase this on master, and re-push, and it should pass CI. Sorry for the hassle on this.

**Review by mcy — APPROVED** [review]

**Review by timothytrippel — APPROVED** [review]

**Pull request merged**